test/higherorderfns.jl

```diff
@@ -8,24 +8,39 @@ module HigherOrderFnsTests
 
 using Test
 using SparseArrays
+using SparseArrays: nonzeros
 using LinearAlgebra
 using Random
 include("forbidproperties.jl")
-
+function test_map_and_map!(A, alloc_tests)
+    # --> test map entry point
+    fA = Array(A)
+    shapeA = size(A)
+    @test map(sin, A) == sparse(map(sin, fA))
+    @test map(cos, A) == sparse(map(cos, fA))
+    # --> test map! entry point
+    fX = copy(fA); X = similar(A)
+    map!(sin, X, A); X = similar(A) # warmup for @allocated
+    map!(sin, X, A); X = similar(A) # warmup for @allocated
+    if @allocated(map!(sin, X, A)) == 0 || alloc_tests
+        println(stderr, "A in test_map_and_map! is ", A)
+    end
```

Comment:
I'm trying to follow the logic here, but fail. IIUC, you want this to print when `alloc_tests == false`(the third call below), which will happen when `@allocated(map!(sin, X, A)) == 0`? Do you mean
```suggestion
    if !(@allocated(map!(sin, X, A)) == 0 || alloc_tests)
        println(stderr, "A in test_map_and_map! is ", A)
    end
```
`alloc_tests = true` sounds to me like "do the allocation tests", whereas you seem to use it as "skip the allocation tests`.